logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageRepository.kt

```diff
@@ -337,8 +337,12 @@ class MessageDataSource(
         }
 
         val messageOption = when (messageTarget) {
-            is MessageTarget.Client -> MessageApi.QualifiedMessageOption.IgnoreAll
+            is MessageTarget.Client.IgnoreIfMissing -> MessageApi.QualifiedMessageOption.IgnoreAll
             is MessageTarget.Conversation -> MessageApi.QualifiedMessageOption.ReportAll
+            is MessageTarget.Client.ReportIfMissing ->
```

Comment:
For federated environments, even tho, we don't have a definition for ephemeral messages. I would incline to use`IgnoreSome` instead, as we did for sending messages in remote backends' context. Mainly because we could be stuck, with users that never (or really late) might come alive.

Reply:
As a side comment, we can adjust later on if we are not clear about it, or it's a big change =) 
Will not make a difference in this current stage

Reply:
you mean add the list of federated users to the ignore some list ?

Reply:
Maybe makes more sense to fix this when merging to develop.

Reply:
@yamilmedina lets sync about this one tomorrow once I raise the same fix against dev

Reply:
Yeah sure, no problem mate, as I mentioned we can do sync later. No need for it now :D